src/features/projectsV2/ProjectSnippet/index.tsx

```diff
@@ -146,20 +158,41 @@ export default function ProjectSnippet({
     showTooltipPopups,
     page,
     project,
+    setPreventShallowPush,
   };
 
   const projectPath = useMemo(() => {
-    let path = `/${locale}/prd/${project.slug}`;
+    let path = `/${locale}/prd/${project.slug}?backNavigationUrl=${router.asPath}`;
+    const params = new URLSearchParams();
+
     if (embed === 'true') {
-      const params = new URLSearchParams({ embed: 'true' });
+      params.append('embed', 'true');
       if (callbackUrl !== undefined && typeof callbackUrl === 'string') {
         params.append('callback', callbackUrl);
       }
-      path += `?${params.toString()}`;
     }
+
+    const paramsString = params.toString();
+    if (paramsString) {
+      path += `?${paramsString}`;
+    }
+
     return path;
-  }, [locale, project.slug, embed, callbackUrl]);
+  }, [locale, project.slug, embed, callbackUrl, page]);
```

Comment:
There's something wrong in the params logic.

If I visit `en/prd?embed=true`, the projectPath gets calculated as `/en/prd/:projectSlug?backNavigationUrl=/en/prd?embed=true?embed=true`.

You probably need to add url encoding to the backNavigationUrl, and also check the logic for adding params so that we have the pattern of `?param1=value1&param2=value2` not `?param1=value1?param2=value2`